pkg/kds/v2/client/kds_client.go

```diff
@@ -112,7 +113,19 @@ func (s *kdsSyncClient) Receive() error {
 		}
 		err = s.callbacks.OnResourcesReceived(received)
 		if err != nil {
-			return errors.Wrapf(err, "failed to store %s resources", received.Type)
+			if store.IsResourceAlreadyExists(err) {
+				s.log.Info("received resource already exists, sending NACK", "err", err)
+				if err := s.kdsStream.NACK(received.Type, err); err != nil {
+					if err == io.EOF {
+						return nil
+					}
+					return errors.Wrap(err, "failed to NACK a discovery response")
+				}
+				s.log.V(1).Info("NACK", "response", received)
+				continue
+			} else {
+				return errors.Wrapf(err, "failed to store %s resources", received.Type)
+			}
```

Comment:
Should we filter this `ResourceAlreadyExists` error when creating and then downgrade the error? I feel confused if we send NACK, will we do this create again in the next streaming?

Reply:
I've changed the logic